### PR TITLE
PrimaryButton with icon + PD Step icons

### LINKED ISSUE #582: Update PrimaryButton to take an optional Icon
## overview

To support the [Protocol Designer step list](https://github.com/Opentrons/opentrons/milestone/2), we need to allow the PrimaryButton to take an optional iconName. This is to support the "Add Step" button(s):

image

We also need to ensure all icons needed for PD step types are available in the components library.

## tasks

*   [x] Add optional iconName prop to PrimaryButton
    *   Icon should render on the far left of the button
*   [ ] Add icons to icon-data for transfer types
    *   Arrow right (Transfer)
    *   Split arrows right (Distribute)
    *   Combine arrows right (Consolidate)
    *   Up down arrows (Mix)
    *   Pause (Pause)
    *   ~Flask (Deck Setup)~ Already in place

### PR DESCRIPTION
**Pull request by IanLondon**
WIP.

fixes #582

### CODE REVIEW

**Comment by IanLondon** [comment]
Also I changed the test for button so `onClick` is set to `undefined` when button is disabled, (instead of being set to `false`), to fix this warning:

```
main.bundle.js:66724 Warning: Expected `onClick` listener to be a function, instead got `false`.

If you used to conditionally omit it with onClick={condition && value}, pass onClick={condition ? value : undefined} instead.
    in button (created by Button)
    in Button (created by PrimaryButton)
```

File: components/src/buttons/buttons.css [hunk 1]
```diff
@@ -48,3 +48,8 @@
   background-color: #d2d2d2;
   box-shadow: none;
 }
+
+.buttonIcon {
```
Comment (mcous) [inline components/src/buttons/buttons.css]: Can we make this camel_case to match everything else?

File: components/src/icons/Icon.md [hunk 2]
```diff
@@ -4,7 +4,8 @@ All available icons:
 // import {ALERT, BACK, REFRESH, etc.} from @opentrons/components
 const {
   ALERT, BACK, CLOSE, REFRESH, SPINNER, USB, WIFI, FLASK, CHECKED, UNCHECKED, CHEVRON_UP,
-  CHEVRON_DOWN, CHEVRON_LEFT, CHEVRON_RIGHT, FILE, COG, CONNECT,
+  CHEVRON_DOWN, CHEVRON_LEFT, CHEVRON_RIGHT, FILE, COG, CONNECT, CONDENSE, DISTRIBUTE,
```
Comment (mcous) [inline components/src/icons/Icon.md]: Aside: after reading through the flow + redux stuff, I wonder if we really need to export all these? As opposed to something like:

```js
// icon-data.js
export type IconName =
  'file' |
  'cog' |
  // etc...
```

File: components/src/buttons/Button.js [hunk 3]
```diff
@@ -25,7 +28,7 @@ export type ButtonProps = {
  */
 export default function Button (props: ButtonProps) {
   const {disabled} = props
-  const onClick = !disabled && props.onClick
+  const onClick = disabled ? undefined : props.onClick
```
Comment (mcous) [inline components/src/buttons/Button.js]: Why was this change made? I get a little uneasy around explicitly setting things / passing `undefined` and React knows that if `false` get passed for `onClick` then it means there's no click handler

File: components/src/icons/Icon.md (hunk 2 above)
Comment (IanLondon) [reply components/src/icons/Icon.md]: I had the same thought after you talked to me about Redux action types... it's a new world with flux

File: components/src/buttons/Button.js (hunk 3 above)
Comment (IanLondon) [reply components/src/buttons/Button.js]: Made to fix ```
main.bundle.js:66724 Warning: Expected `onClick` listener to be a function, instead got `false`.

If you used to conditionally omit it with onClick={condition && value}, pass onClick={condition ? value : undefined} instead.
    in button (created by Button)
    in Button (created by PrimaryButton)`
   ...
```

**Pull request merged**